plugins/org.genivi.faracon/src/org/genivi/faracon/franca2ara/ARATypeCreator.xtend

```diff
@@ -236,11 +243,61 @@ class ARATypeCreator extends Franca2ARABase {
 	}
 
 	def private dispatch create fac.createImplementationDataType createDataTypeForReference(FArrayType fArrayType) {
+		// Request deployment data for the given array data type.
```

Comment:
I guess everything that belongs to the comment "Request deployment data for the given array data type" should be in an own method.

Reply:
I extracted the whole deployment data access required for fixed sized arrays into a separate helper class.